tests/e2e/specs/api/NodeJsExpressDevFileAPI.spec.ts

```diff
@@ -0,0 +1,155 @@
+
+		if (BASE_TEST_CONSTANTS.IS_CLUSTER_DISCONNECTED()) {
+			Logger.info('Test cluster is disconnected. Init Java Truststore...');
+			const initJavaTruststoreCommand: string =
+				'cp /home/user/init-java-truststore.sh /tmp && chmod +x /tmp/init-java-truststore.sh && /tmp/init-java-truststore.sh';
+			const output: ShellString = containerTerminal.execInContainerCommand(initJavaTruststoreCommand, containerName);
+			expect(output.code).eqls(0);
+		}
+
+		const workdir: string = YAML.parse(devfileContent).commands[0].exec.workingDir;
+		const commandLine: string = YAML.parse(devfileContent).commands[0].exec.commandLine;
+		Logger.info(`workdir from exec section of DevWorkspace file: ${workdir}`);
+		Logger.info(`commandLine from exec section of DevWorkspace file: ${commandLine}`);
+
+		let runCommandInBash: string = commandLine.replaceAll('$', '\\$'); // don't wipe out env. vars like "${PROJECTS_ROOT}"
+		if (workdir !== undefined && workdir !== '') {
+			runCommandInBash = `cd ${workdir} && ` + runCommandInBash;
+		}
+
+		const output: ShellString = containerTerminal.execInContainerCommand(runCommandInBash, containerName);
+		expect(output.code).eqls(0);
+
+		const outputText: string = output.stdout.trim();
+		expect(outputText).contains('Run `npm audit` for details.');
+	});
+
+	test('Check "run the web app" command', function (): void {
+		const containerName: string = YAML.parse(devfileContent).commands[0].exec.component;
+		const workdir: string = YAML.parse(devfileContent).commands[1].exec.workingDir;
+		const commandLine: string = YAML.parse(devfileContent).commands[1].exec.commandLine;
+		Logger.info(`workdir from exec section of DevWorkspace file: ${workdir}`);
+		Logger.info(`commandLine from exec section of DevWorkspace file: ${commandLine}`);
+
+		let runCommandInBash: string = commandLine.replaceAll('$', '\\$'); // don't wipe out env. vars like "${PROJECTS_ROOT}"
+		if (workdir !== undefined && workdir !== '') {
+			runCommandInBash = `cd ${workdir} && ` + runCommandInBash;
+		}
+
+		const output: ShellString = containerTerminal.execInContainerCommandWithTimeout(runCommandInBash, containerName);
+		expect(output.code).eqls(124);
+
+		const outputText: string = output.stdout.trim();
+		expect(outputText).contains('Example app listening on port 3000!');
+	});
+
+	test('Check "stop the web app" command', function (): void {
+		const containerName: string = YAML.parse(devfileContent).commands[0].exec.component;
+		const workdir: string = YAML.parse(devfileContent).commands[4].exec.workingDir;
+		const commandLine: string = YAML.parse(devfileContent).commands[4].exec.commandLine;
+		Logger.info(`workdir from exec section of DevWorkspace file: ${workdir}`);
+		Logger.info(`commandLine from exec section of DevWorkspace file: ${commandLine}`);
+
+		// Prettier changes next line to `replaceAll("'", "'\"'\"'")` that throws an error from eslint.
+		// prettier-ignore
+		let runCommandInBash: string = commandLine.replaceAll('\'', '\'\"\'\"\'');
```

Comment:
it would be useful to see the actual "stop the web app" command in the comment, which is needed to be adjusted to the run by oc exec command

Reply:
Updated

Reply:
thanks